docs/specification/draft/basic/authorization.mdx

```diff
@@ -267,35 +257,23 @@ Servers **MUST** return appropriate HTTP status codes for authorization errors:
-
-## 3. Best Practices
 
-#### 3.1 Local clients as Public OAuth 2.1 Clients
+## 3. Security Considerations
 
-We strongly recommend that local clients implement OAuth 2.1 as a public client:
+### 3.1 Client Token Theft
+An attacker who gains access to a client's stored tokens can make unauthorized requests to resource servers. Clients MUST securely store tokens following OAuth 2.0 best practices.
 
-1. Utilizing code challenges (PKCE) for authorization requests to prevent interception
-   attacks
-2. Implementing secure token storage appropriate for the local system
-3. Following token refresh best practices to maintain sessions
-4. Properly handling token expiration and renewal
+### 3.2 Server Token Theft
+An attacker who compromises an MCP authorization server may access stored tokens. MCP authorization servers SHOULD enforce token expiration and rotation to limit the window of exploitation.
 
-#### 3.2 Authorization Metadata Discovery
+### 3.3 Token Interception
+An attacker positioned between clients and servers can intercept tokens transmitted over insecure connections. All authorization endpoints MUST be served over HTTPS. Clients MUST implement PKCE according to [OAuth 2.1 section 7.5.2](https://www.ietf.org/archive/id/draft-ietf-oauth-v2-1-12.html#name-countermeasures)
 
-We strongly recommend that all clients implement metadata discovery. This reduces the
-need for users to provide endpoints manually or clients to fallback to the defined
-defaults.
+### 3.4 Open Redirection
+An attacker may craft malicious redirect URIs to direct users to phishing sites. MCP authorization servers MUST validate redirect URIs against pre-registered values to prevent redirection attacks.
 
-#### 3.3 Dynamic Client Registration
+###  3.5 Insecure Redirect URIs
+An attacker can capture data transmitted to non-secure endpoints. Redirect URIs MUST be either localhost URLs or HTTPS URLs to prevent token and code interception.
 
-Since clients do not know the set of MCP servers in advance, we strongly recommend the
-implementation of dynamic client registration. This allows applications to automatically
-register with the MCP server, and removes the need for users to obtain client ids
-manually.
+### 3.6 Confused Deputy Problem
+An attacker can exploit OAuth proxy configurations that share 3rd party client credentials across multiple users. When an MCP server fronts another authorization server that does not support dynamic client registration, the MCP uses a static client_id with the backing service. If the backing service sets cookies after user authorization, an attacker can craft malicious authorization requests that bypass consent screens for previously authorized applications. MCP servers using a static client_id for a backing service MUST require explicit approval for each newly registered dynamic client prior to forwarding requests to the backing authorization server for user consent.
```

Comment:
```suggestion
An attacker can exploit OAuth proxy configurations that share third-party client credentials 
across multiple users.

When an MCP server fronts an authorization server that does not support dynamic client registration, the MCP server will use a static client ID to acquire credentials for the 
upstream API.

If the MCP server sets cookies after user authorization, an attacker can craft malicious authorization requests that bypass consent flows for previously authorized applications.

MCP servers using a static client ID for an upstream API **MUST** require explicit approval for each newly dynamically registered client prior to forwarding requests to the backing authorization server for user consent.
```

Reply:
> If the MCP server sets cookies after user authorization
It's the 3p / backing AS that sets the cookie in this case, see this issue: https://github.com/modelcontextprotocol/modelcontextprotocol/discussions/265 

will do the other adjustments